### DISCUSSION TITLE
Delete branch cache on branch deletion

### DISCUSSION THREAD

**Discussion by rdlf0** [community/community#38807] [Actions]
### Select Topic Area

Product Feedback

### Body

Currently, when a PR gets merged and its branch gets (auto)deleted, the cache associated with this branch gets reassigned to master/main. I don't see a point in keeping a cache for a non-existing branch. I'd like to see an option to also delete caches for a branch when it gets deleted.

For example, I have a workflow triggered on `pull_request: opened/reopened/synchronize` that builds my Maven-based Java project and caches the Maven's dependencies with the `setup-java` action in order to avoid fetching them on every push to the PR. Once I merge my PR, my branch gets auto-deleted. From this point I don't need that cache anymore, so it is safe for it to be deleted, too.

**Comment by ossanna16** [comment]
Hi there @rdlf0 and welcome to our community! Thank you for the feedback!
